### PR TITLE
Refine documentation for tuning and indexing to optimize performance based on PostgreSQL and Eclipselink enhancements.

### PR DESCRIPTION
**Pull request by tapaswenipathak**
Ref: https://github.com/openMF/mifos-documentation/issues/14

TODO:

- [ ] Describing tuning and indexing based on Eclipselink enhancements.

cc @edcable.

## Description

Describe the changes made and why they were made.

Ignore if these details are present on the associated [Apache Fineract JIRA ticket](https://github.com/apache/fineract/pull/1284).

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [ ] Write the commit message as per https://github.com/apache/fineract/#pull-requests

- [ ] Acknowledge that we will not review PRs that are not passing the build _("green")_ - it is your responsibility to get a proposed PR to pass the build, not primarily the project's maintainers.

- [ ] Create/update unit or integration tests for verifying the changes made.

- [ ] Follow coding conventions at https://cwiki.apache.org/confluence/display/FINERACT/Coding+Conventions.

- [ ] Add required Swagger annotation and update API documentation at fineract-provider/src/main/resources/static/legacy-docs/apiLive.htm with details of any API changes

- [ ] Submission is not a "code dump".  (Large changes can be made "in repository" via a branch.  Ask on the developer mailing list for guidance, if required.)

FYI our guidelines for code reviews are at https://cwiki.apache.org/confluence/display/FINERACT/Code+Review+Guide.

### CODE REVIEW

File: fineract-doc/src/docs/en/chapters/architecture/fineract-database-handling.adoc [hunk 1]
```diff
@@ -0,0 +1,114 @@
+= Fineract Database Handling
```
Comment (galovics) [inline fineract-doc/src/docs/en/chapters/architecture/fineract-database-handling.adoc]: Rather call it Database Support

File: fineract-doc/src/docs/en/chapters/architecture/fineract-database-handling.adoc [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+== Fineract Database Upgrades
```
Comment (galovics) [inline fineract-doc/src/docs/en/chapters/architecture/fineract-database-handling.adoc]: What's this section title, upgrades? Upgrades of what? Below the text is about the different database supports.

File: fineract-doc/src/docs/en/chapters/architecture/fineract-database-handling.adoc [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+The platform differentiates between these database types in certain cases when
+there's a need to use some database specific tooling. To do so, the platform
+examines the JDBC driver used for running the platform and tries to determine
+which database is being used.
+
+Fineract doesn't guarantee a backward compatible database upgrades.
+
+=== Background
+
+The Mifos team has stated that non-backward compatible database upgrades are
```
Comment (galovics) [inline fineract-doc/src/docs/en/chapters/architecture/fineract-database-handling.adoc]: I don't think we should say "The Mifos team". This is a community driven project under the Apache umbrella, not a Mifos driven project.

**Review by galovics — CHANGES_REQUESTED** [review]

File: fineract-doc/src/docs/en/chapters/architecture/fineract-database-handling.adoc (hunk 3 above)
Comment (tapaswenipathak) [reply fineract-doc/src/docs/en/chapters/architecture/fineract-database-handling.adoc]: i don't think we should add this doc at all as customer documenation. More of a planning and ideas doc. we need to discuss this on where and why this doc can find a place. cc @edcable.

**Review by jdailey — CHANGES_REQUESTED** [review]
This case study format doesn't belong in documentation.  Mifos does not make decisions for the Fineract project.  Documentation should focus on scalability requirements ... gathered by various fineract volunteers.

File: fineract-doc/src/docs/en/chapters/architecture/fineract-database-handling.adoc (hunk 3 above)
Comment (edcable) [reply fineract-doc/src/docs/en/chapters/architecture/fineract-database-handling.adoc]: @tapaswenipathak as you have noted and as @jdailey suggested, this documentation shouldn't include any of the rationale based on customer's needs of why the changes were made. You are both correct that this portion of the documentation shouldn't include all of this background which almost seems like a case study. 

Regarding @galovics comment, we should refer to it being a community-driven project under Apache umbrella but to give you context Arnold, these documentation-related efforts are all purely Mifos-driven as part of our Google Season of Docs project that we secured with support from Google. 

Intent of this section were to note the changes that were made to support PostgreSQL and Eclipselink, (to the extent that we can), the changes that needed to be made to support PostgreSQL vs MySQL/MariaDB, the process for developers/implementers to upgrade to benefit from these changes and then to specifically document the tuning and indexing specific to PostgreSQL to achieve optimal performance.

File: fineract-doc/src/docs/en/chapters/architecture/persistence.adoc [hunk 4]
```diff
@@ -62,3 +62,129 @@ The switch from Flyway (1.6.x) to Liquibase (1.7.x) was planned to be as smooth
 * If the database is empty, Liquibase will create the database schema from scratch
 * If the database contains the latest Fineract 1.6.x database structure which was previously migrated with Flyway. Liquibase will seamlessly upgrade it to the latest version. Note: the Flyway related 2 database tables are left as they are and are not deleted.
 * If the database contains an earlier version of the database structure than Fineract 1.6.x. Liquibase will *NOT* do anything and *will fail the application during startup*. The proper approach in this case is to first upgrade your application version to the latest Fineract 1.6.x so that the latest Flyway changes are executed and then upgrade to the newer Fineract version where Liquibase will seamlessly take over the database upgrades.
+
+== Tuning and Indexing
+
+To optimize performance based on PostgreSQL and Eclipselink enhancements, tuning
```
Comment (tapaswenipathak) [inline fineract-doc/src/docs/en/chapters/architecture/persistence.adoc:68]: cc @mdallos.

**Pull request closed without merging**